webhooks/pod_webhook.go

```diff
@@ -244,6 +248,63 @@ func (m *PodMutator) injectSidecar(
 	return json.Marshal(pod)
 }
 
+func (m *PodMutator) isKubeProxyReady(ctx context.Context) (bool, bool, error) {
+	m.Client.Scheme()
+	d := appsV1.Deployment{}
+	err := m.Client.Get(ctx, client.ObjectKey{Name: controllers.KubeProxyDeploymentName, Namespace: m.KubeProxyConfig.Namespace}, &d)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// does not exist, is not ready, no error
+			return false, false, nil
+		}
+		// does not exist, is not ready, is in error
+		return false, false, err
+	}
+	if d.Status.ReadyReplicas == 0 {
+		// exists, not ready, no error
+		if d.CreationTimestamp.Time.Before(time.Now().Add(-3 * time.Minute)) {
+			return true, false, fmt.Errorf(
+				"flagd-kube-proxy not ready after 3 minutes, was created at %s",
+				d.CreationTimestamp.Time.String(),
+			)
+		}
+		return true, false, nil
+	}
+	// exists, at least one replica ready, no error
+	return true, true, nil
+}
+
+func (m *PodMutator) handleKubeProxy(ctx context.Context, sidecar *corev1.Container, source v1alpha1.Source) error {
+	// does the proxy exist
+	exists, ready, err := m.isKubeProxyReady(ctx)
+	if err != nil {
+		return err
+	}
+	if !exists || (exists && !ready) {
+		return &kubeProxyDeferError{}
+	}
+
+	config := []types.SourceConfig{
+		{
+			Provider: "grpc",
+			Selector: fmt.Sprintf("core.openfeature.dev/%s", source.Source),
```

Comment:
I _think_ this needs to be changed to prefix the namespace of the source, if not present. We may want a util function for this since I think the k8s sync is doing something similar.

See: https://github.com/open-feature/open-feature-operator/pull/412#issuecomment-1487439625

Reply:
resolved